spec/rspec/support/differ_spec.rb

```diff
@@ -261,6 +261,35 @@ def inspect; "<BrokenObject>"; end
           expect(diff).to be_diffed_as(expected_diff)
         end
 
+        it "outputs unified diff message of two hashes with BigDecimal keys" do
+          in_sub_process_if_possible do
+            require 'bigdecimal'
+            expect {
+              diff = differ.diff({ BigDecimal('3') => 'b'}, { 3 => 'b' })
+              if expect(diff).to match(/3.0 \(#<BigDecimal/)
+                raise StandardError, 'Passed Test'
+              end
+            }.to raise_error(StandardError, 'Passed Test')
```

Comment:
Why are you using `expect` in a conditional, then raising an error, then expecting that error to get raised?  It seems quite convoluted compared to:

``` ruby
diff = differ.diff({ BigDecimal('3') => 'b'}, { 3 => 'b' })
expect(diff).to match(/3.0 \(#<BigDecimal/)
```